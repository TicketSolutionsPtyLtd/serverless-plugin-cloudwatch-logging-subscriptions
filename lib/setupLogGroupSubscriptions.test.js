'use strict'

const sinon = require('sinon')
const Serverless = require('serverless')

describe('Setup LogGroup Subscriptions', () => {
    const serverless = new Serverless()
    serverless.cli = {
        log: sinon.spy(),
        consoleLog: sinon.spy(),
    }

    serverless.service.provider.compiledCloudFormationTemplate = {
        Resources: {},
    }

    serverless.service.functions = {
        testing: {
            name: 'testing',
        },
    }

    const baseModule = require('./setupLogGroupSubscriptions') // eslint-disable-line global-require

    const module = Object.assign({
        serverless,
        options: {},
    }, baseModule)

    it('should setup log group subscriptions', () => {
        module.setupLogGroupSubscriptions()
        expect(serverless.service.provider.compiledCloudFormationTemplate.Resources)
            .toMatchSnapshot()
    })
})
