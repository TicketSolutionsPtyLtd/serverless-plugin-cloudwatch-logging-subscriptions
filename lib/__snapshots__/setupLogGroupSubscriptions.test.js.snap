exports[`Setup LogGroup Subscriptions should setup log group subscriptions 1`] = `
Object {
  "LoggingLambdaPermission": Object {
    "Properties": Object {
      "Action": "lambda:InvokeFunction",
      "FunctionName": Object {
        "Fn::GetAtt": Array [
          "LogHandlerLambdaFunction",
          "Arn",
        ],
      },
      "Principal": Object {
        "Fn::Join": Array [
          "",
          Array [
            "logs.",
            Object {
              "Ref": "AWS::Region",
            },
            ".amazonaws.com",
          ],
        ],
      },
    },
    "Type": "AWS::Lambda::Permission",
  },
  "TestingLambdaFunctionSubscriptionFilter": Object {
    "DependsOn": "LoggingLambdaPermission",
    "Properties": Object {
      "DestinationArn": Object {
        "Fn::GetAtt": Array [
          "LogHandlerLambdaFunction",
          "Arn",
        ],
      },
      "FilterPattern": "",
      "LogGroupName": "/aws/lambda/testing",
    },
    "Type": "AWS::Logs::SubscriptionFilter",
  },
}
`;
